### PR TITLE
Onboarding Improvements: Extract Or Layout

### PR DESCRIPTION
**Pull request by ParaskP7**
This PR extracts the or layout to be later on easily reused. On the newly created shared `login_or_layout` bottom and top padding were added based on its overall usage. This way, every client that is to be using this newly created shared `login_or_layout` will conform to its now defined standards.

Note: As part of this change the `6dp` login layout `layout_marginBottom` was replaced by the inner `8dp` or layout
`paddingBottom` and `paddingTop`. The `paddingTop` change is irrelevant since the `login_email_screen` layout will anyway adjust that automatically. However, the 'paddingBottom' change will have a diff of `2dp`. This change can be also considered as negligible, since it is almost unnoticed.

### CODE REVIEW

**Review by ashiagr — APPROVED** [review]
Thanks for extracting "or" layout 🙏. Everything looks good code-wise. I'll leave the decision on @osullivanchris about the bottom `2dp` extra padding. Also, let's wait for the CI issue to be resolved. Feel free to merge thereafter.

**Comment by ParaskP7** [comment]
👋 @ashiagr !

Thanks so much for this review! 🙏 

> Thanks for extracting "or" layout 🙏. Everything looks good code-wise. I'll leave the decision on @osullivanchris about the bottom 2dp extra padding. Also, let's wait for the CI issue to be resolved. Feel free to merge thereafter.

Yes, before I merge this PR I'll first incorporate this change within [WordPress-Android#15036](https://github.com/wordpress-mobile/WordPress-Android/pull/15036), using a hash version, get an approval from @osullivanchris and then move forward.

**Comment by ParaskP7** [comment]
👋 @ashiagr !

A quick update from my side.

> Yes, before I merge this PR I'll first incorporate this change within WordPress-Android#15036, using a hash version, get an approval from @osullivanchris and then move forward.

After getting the 👍 from @osullivanchris on the `Login Email` screen and the `Or Layout` above the `Google Sign-In` button, see [here](https://github.com/wordpress-mobile/WordPress-Android/pull/15036#discussion_r673201584), I am now going to proceed with merging this PR.

**Pull request merged**
